core/commands/inline_diff.py

```diff
@@ -343,8 +340,7 @@ class GsInlineDiffFocusEventListener(EventListener):
     latest file status when the view regains focus.
     """
 
-    def on_activated(self, view):
-
+    def on_activated_async(self, view):
         if view.settings().get("git_savvy.inline_diff_view") is True:
             view.run_command("gs_inline_diff_refresh")
 
```

Comment:
That's a mistake. If you run a command like `view.run_command("gs_inline_diff_refresh")` it always starts at the main thread.

Look at how it is done for the diff view 

https://github.com/divmain/GitSavvy/blob/29642a3f5731be4cbd63f9acd56b4b714fbc0001/core/commands/diff.py#L129-L138

`sync` being the default you call it with `sync=False` in on_activated and in the main gs_inline_diff command.

Reply:
Do you know if there is a reliable way to add sleep to be blocking so I can check it works like I expect

Reply:
Don't know what you're trying to do. `time.sleep` is always blocking. It is also important to check on which thread you run. (E.g. `import threading; print threading.current_thread()`)

Reply:
Take a look now